File: 7.原型模式/geqiangbao_20201209_cloneable_oc/geqiangbao_20201209_cloneable_oc/原型模式/Person.m
//
//  Person.m
//  geqiangbao_20201209_cloneable_oc
//
//  Created by xiaobao on 2020/12/9.
//

#import "Person.h"

@implementation Person

- (id)clone{
    Person *p1 = [[Person alloc]init];
    p1.name = self.name;
    p1.age = self.age;
    return p1;
    
}

@end

File: 7.原型模式/geqiangbao_20201126_cloneable_swift/geqiangbao_20201126_cloneable_swift/demo2/Person.swift
//
//  Person.swift
//  geqiangbao_20201126_cloneable_swift
//
//  Created by xiaobao on 2020/11/26.
//

import UIKit

class Person: NSObject,CloneableProtocol {
    var name:String
    var age:Int
    
    init(name:String,age:Int) {
        self.name = name
        self.age = age
    }
    
    func clone() -> Person {
        return Person(name: self.name, age: self.age)
    }
    
    

}
